### DISCUSSION TITLE
Handling requests that should return different results on the second call

### DISCUSSION THREAD

**Discussion by fakeleft** [vcr/vcr#869] [Ideas]
This may not be so much an issue as it is a help/suggestion request.

I'm trying to use VCR to test my Stripe integration. If I run the following example sequence of calls:

``` ruby
p = Stripe::Plan.retrieve("some_plan")
p.delete
Stripe::Plan.retrieve("some_plan")
```

Then the last line should throw a Stripe::InvalidRequestError, as the plan no longer exists. Of course with VCR, I incorrectly get back the original plan instead, since that's what got recorded the first time retrieve was called. I can't see any way of tweaking the request matching to handle this, and I'm surprised I'm failing to google my way to someone else encountering this problem. One way I can think of maybe doing it is by having separate cassettes for lines 1 and 3, and switching between them in my test. Any other suggestions?

**Comment by nbibler** [comment]
Interesting. I don't _think_ that's how VCR should be working and it likely depends on how you recorded your cassette.  Generally, during the recording phase, VCR will record _all_ interactions in the proper order. During playback, they'll all be executed back in the same order they were recorded.

So, presumably, if you made the retrieve call, followed by a delete, followed by a retrieve, you should see three requests in your recorded cassette.

This depends entirely on your matcher configuration, cassette recording method, and whether or not the recording was run prior to the addition of the third line of code or not.

Does that help?

**Comment by sigmavirus24** [comment]
> I don't think that's how VCR should be working and it likely depends on how you recorded your cassette.

Having implemented VCR in a different language, that's it works. If you do

```
GET /foo
DELETE /bar
GET /foo
```

The second `GET /foo` will replay the first one. If I remember correctly, `record: :all` doesn't do what it sounds like either.

**Comment by fakeleft** [comment]
In case anyone runs into this, I had some success with cassette switching, but keeping track of what state things were in after multiple requests/cassette switches became pretty intractable, especially given that the best solution for this seems to be [stripe-ruby-mock](https://github.com/rebelidealist/stripe-ruby-mock). Issue can be closed.

**Comment by krainboltgreene** [comment]
Sounds like this could definitely be turned into a PR?
